### ISSUE TITLE
Weird behavior of asset check for dbt python model

### ISSUE THREAD

**Issue by dduong1603** [dagster-io/dagster#18386]
### Dagster version

1.5.9

### What's the issue?

For context, I'm running dbt python model with `dbt-fal`.

When a job execute both the materialization and the asset checks runs, somehow the asset checks are skipped with this info message `op 'dbt_python_assets' did not yield or return expected outputs`. However, if I run the asset checks alone (using the Execute All in the Checks tab), the checks run as expected, using the same dbt command. The same dbt command runs normally in the terminal as well, and I don't see this behavior with a regular dbt SQL model, so I'm not really sure how to debug this.

The code to define the python asset is below. It's the same code as a regular dbt SQL asset, just with different profile target.
```python
@dbt_assets(
    manifest=DBT_DAGSTER_MANIFEST,
    select="tag:fal",
    exclude="resource_type:seed",
    dagster_dbt_translator=dagster_global_dbt_translator, # enable asset check
)
def dbt_python_assets(
    context: OpExecutionContext,
    dbt: DbtCliResource,
):
    yield from dbt.cli(
        ["run", "--target", f"{deployment_name}-with-fal"], context=context
    ).stream()
    yield from dbt.cli(
        ["test", "--target", f"{deployment_name}-with-fal"], context=context
    ).stream()
```

### What did you expect to happen?

Asset check can run successfully (not skipped) in the same run as asset materialization

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by johannkm** [comment]
@dduong1603  Did this happen to occur only when executing all models and tests in `dbt_python_assets`? A fix is going out today in release 1.5.10 for a bug where checks wouldn't run when not doing a subset: https://github.com/dagster-io/dagster/pull/18117

**Comment by dduong1603** [comment]
> @dduong1603 Did this happen to occur only when executing all models and tests in `dbt_python_assets`? A fix is going out today in release 1.5.10 for a bug where checks wouldn't run when not doing a subset: #18117

oh interesting. yeah `dbt_python_assets` currently only has 1 asset so no subsetting was done on it. I'll keep an eye on for 1.5.10 to test out. Thank you!

**Comment by dduong1603** [comment]
@johannkm happy to say that the behavior is now correct with 1.5.10 :) thank you!
